pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -628,6 +622,60 @@ func upgradeEligibilityCheck(upgrade *harvesterv1.Upgrade, repoInfo *RepoInfo) (
 	return true, ""
 }
 
+func (h *upgradeHandler) prepareNodesForUpgrade(upgrade *harvesterv1.Upgrade, repoInfoStr string) (*harvesterv1.Upgrade, error) {
+	upgradeConfig, err := settings.DecodeConfig[settings.UpgradeConfig](settings.UpgradeConfigSet.Get())
```

Comment:
Correct me if I'm wrong, but we need to get upgradeConfig through settingsCache, like https://github.com/harvester/harvester/blob/ccc71be7225ce76697da2bbe8aeb14507e1dbd5b/pkg/api/vm/handler.go#L606, otherwise this line always get default value.

Reply:
Please ignore this noise, the Provider inside settings already include SettingsCache, which means the Get method already done what I mentioned in the above code.